docs: clarify usage of `createContext()` (#3799)
trpc/trpc: www/docs/server/context.md

## Creating the context

> The `createContext()` function must be passed to whichever handler is mounting your appRouter, which may be via HTTP, a [server-side call](server-side-calls) or by using our [SSG helper](ssg-helpers).

Reviewer: ```suggestion
The `createContext()` function must be passed to the handler that is mounting your appRouter, which may be via HTTP, a [server-side call](server-side-calls) or our [SSG helper](ssg-helpers).
```

> `createContext()` is called for each call to a procedure.

Reviewer: This isn't actually true if you're using batching...we should probably clarify that it's called for "each request to tRPC" (or something like that)

Author: See what you make of that. I'm thinking "invocation" since server-side it may not always be accurate to refer to a 'request' since it connotes a network being involved?

Reviewer 2: @sachinraja any further comments here?

Reviewer: invocation sounds good :+1:
